src/services/AbstractPalletsService.ts

```diff
@@ -234,13 +238,19 @@ export abstract class AbstractPalletsService extends AbstractService {
 				palletItemIdx,
 				palletItemId
 			);
-		} else {
+		} else if (metadataFieldType === 'events') {
 			[palletItemIdx, palletItemMeta] = this.getEventItemMeta(
 				historicApi,
 				palletMeta as PalletMetadataV14,
 				palletItemIdx,
 				palletItemId
 			);
+		} else {
```

Comment:
I see that this `if...else` statement keeps growing and I was wondering if it would a good idea to replace this conditionals with a lookup table or an object lookup ? since the only thing that changes is the name of the function called. 
I was checking some articles to find as references : 
- [js patterns - lookup tables](https://omwri.medium.com/javascript-patterns-lookup-tables-26bbaf693e24)
- [js - objects for lookups](https://levelup.gitconnected.com/javascript-algorithm-using-objects-for-lookups-d28f34a59504)
- [js ts code - lookups](https://dev.to/k_penguin_sato/use-lookup-tables-for-cleaning-up-your-js-ts-code-9gk)

Maybe it makes it also more readable and maintainable ? Just an idea because I used that in C in the past and I think it is quite fast but I am not sure in Typescript (performance wise).

Reply:
I don't think this is necessary. A lookup table would require a few abstractions, and performance isn't really going to get affected here. `if...else` is acceptable IMO. We know the fixed size of this if else so I don't think its going to grow much more. 

That being said the only other option I would see for readability is a `switch` statement.